src/Avalonia.Controls/Primitives/UniformGrid.cs

```diff
@@ -102,7 +140,13 @@ protected override Size ArrangeOverride(Size finalSize)
                     continue;
                 }
 
-                child.Arrange(new Rect(x * width, y * height, width, height));
+                var rect = new Rect(
+                    x * (width + ColumnSpacing),
```

Comment:
Perf nit: consider caching `ColumnSpacing` and `RowSpacing` to local variables before the loop. While accessing `StyledProperty` values is very optimized, it's still an order of magnitude slower than a standard field.